extensions/arc/src/ui/dashboards/postgres/postgresExtensionsPage.ts

```diff
@@ -182,44 +188,69 @@ export class PostgresExtensionsPage extends DashboardPage {
 				this.disposables.push(
 					dropExtensionsButton.onDidClick(async () => {
 						try {
-							await vscode.window.withProgress(
-								{
-									location: vscode.ProgressLocation.Notification,
-									title: loc.updatingInstance(this._postgresModel.info.name),
-									cancellable: false
-								},
-								async (_progress, _token): Promise<void> => {
-									let index = this.extensionNames.indexOf(e.name, 0);
-									this.extensionNames.splice(index, 1);
-
-									await this._azdataApi.azdata.arc.postgres.server.edit(
-										this._postgresModel.info.name,
-										{
-											extensions: this.extensionNames.join()
-										},
-										this._postgresModel.controllerModel.azdataAdditionalEnvVars);
-
-									try {
-										await this._postgresModel.refresh();
-									} catch (error) {
-										vscode.window.showErrorMessage(loc.refreshFailed(error));
-									}
-								}
-							);
-
-							vscode.window.showInformationMessage(loc.instanceUpdated(this._postgresModel.info.name));
-
+							dropExtensionsButton.enabled = false;
+							await this.dropExtension(e.name);
 						} catch (error) {
-							vscode.window.showErrorMessage(loc.instanceUpdateFailed(this._postgresModel.info.name, error));
+							vscode.window.showErrorMessage(loc.updateExtensionsFailed(error));
+						} finally {
+							dropExtensionsButton.enabled = true;
 						}
 					})
 				);
 
+				if (e.name === 'citus') {
+					dropExtensionsButton.enabled = false;
+				}
+
 				return [e.name, dropExtensionsButton];
 			});
 		}
 	}
 
+	/** Drop extension */
```

Comment:
Comments typically should follow the standard JS Doc format (separate lines and contain param/return value tags)